File: CleverFerretV2/core/database/src/main/java/com/cleverferret/v2/core/database/schema/SyncStateV1.java
package com.cleverferret.v2.core.database.schema;

import com.cleverferret.v2.core.common.VersionedContract;

/**
 * Canonical reconciliation state between local DB and external/cloud providers.
 */
public record SyncStateV1(
        String syncStateId,
        String entityType,
        String entityId,
        String provider,
        String remoteId,
        String deviceId,
        long localVersion,
        long remoteVersion,
        SyncStatusV1 status,
        String conflictStrategy,
        Long lastSyncedAtEpochMs,
        long updatedAtEpochMs,
        String errorMessage
) implements VersionedContract {
    @Override
    public String contractVersion() {
        return "v1";
    }
}

File: CleverFerretV2/core/database/src/main/java/com/cleverferret/v2/core/database/schema/SyncStateV1.kt
package com.cleverferret.v2.core.database.schema

import com.cleverferret.v2.core.common.VersionedContract

data class SyncStateV1(
    val syncStateId: String,
    val entityType: String,
    val entityId: String,
    val provider: String,
    val remoteId: String,
    val deviceId: String,
    val localVersion: Long,
    val remoteVersion: Long,
    val status: SyncStatusV1,
    val conflictStrategy: String,
    val lastSyncedAtEpochMs: Long?,
    val updatedAtEpochMs: Long,
    val errorMessage: String
) : VersionedContract {
    override fun contractVersion(): String = "v1"
}
